api/main.go

```diff
@@ -115,21 +117,28 @@ func main() {
 		server := fireball.NewApp(routes)
 		server.ErrorHandler = controllers.ErrorHandler
 
+		lock := lock.NewDynamoLock(session, cfg.DynamoLockTable(), time.Minute*5)
+
 		// todo: get num workers from config
-		jobTicker := job.RunWorkersAndDispatcher(2, jobStore, jobRunner)
+		jobTicker := job.RunWorkersAndDispatcher(2, jobStore, jobRunner, lock)
 		defer jobTicker.Stop()
 
-		scalerTicker := scalerDispatcher.RunEvery(time.Minute * 5)
-		defer scalerTicker.Stop()
+		sdFN := scaler.NewDaemonFN(jobStore, environmentProvider)
```

Comment:
This is really cool. Just want to make sure my understanding is correct: all daemon-like processes would run via the `daemon` package async, and they can all be initialized with their own periodicity, custom error, and locking mechanism(s).

Reply:
Yes for periodicity and locking, but they don't really have custom errors other than the name that gets logged to stdout, e.g. we can get something like:
```
[ERROR] [TagDaemon] Failed to list tags
[ERROR] [JobDaemon] Failed to delete job 1234
```